Compiler/src/timing.jl

```diff
@@ -1,4 +1,43 @@
 if ccall(:jl_timing_enabled, Cint, ()) != 0
+    if !isdefined(@__MODULE__, :task_local_storage)
+        # During bootstrapping we assume that things run non-concurrently and it is therefore
+        # safe to use a global variable to store the current timing block.
+        const CURRENT_TIMING_BLOCK = RefValue(C_NULL)
+        @inline _get_current_timing_block_nocheck() = CURRENT_TIMING_BLOCK
```

Comment:
```suggestion
        @inline _get_current_timing_block_nocheck() = CURRENT_TIMING_BLOCK[]
```

Do we need the implementation during bootstrap, or can we just make this post-bootstrap only?

Reply:
We might not need to add any zone texts during bootstrap so let's go with post-bootstrap only then.